src/global-firestore-data/semesters.ts

```diff
@@ -1,31 +1,14 @@
-import { SeasonOrdinal } from '../utilities';
-import store from '../store';
 import { semestersCollection } from '../firebase-frontend-config';
+import store from '../store';
 import { GTag, GTagEvent } from '../gtag';
+import { sortedSemesters } from '../utilities';
 
 import {
   addCourseToSelectableRequirements,
   deleteCourseFromSelectableRequirements,
   deleteCoursesFromSelectableRequirements,
 } from './selectable-requirement-choices';
 
-// compare function for FirestoreSemester to determine which comes first by year and type/season
-export const compareFirestoreSemesters = (a: FirestoreSemester, b: FirestoreSemester): number => {
-  if (a.type === b.type && a.year === b.year) {
-    return 0;
-  }
-  if (a.year > b.year) {
-    return -1;
-  }
-  if (a.year < b.year) {
-    return 1;
-  }
-  if (SeasonOrdinal[a.type] < SeasonOrdinal[b.type]) {
-    return 1;
-  }
-  return -1;
-};
```

Comment:
We can get rid of this; `sortedSemesters()` has the same purpose. The only difference is that we don't modify the array directly anymore (think `sorted()` vs `.sort()`, which is good to avoid side effects & when working with `readonly` arrays.